### ISSUE TITLE
Persist Checkout Branch

### ISSUE THREAD

**Issue by jch5f** [cruft/cruft#54]
Currently, if the `checkout` option is used with `create` to select a specific branch when expanding a template there is no way to persist the chosen branch. To stay on the chosen branch, all future `check` and `update` commands require the branch to be manually added with the `checkout` option. I propose  the branch is sored in the `.cruft.json` file. 

During the execution of the `create` or `link` commands, if the `checkout` option is used, then the text of the option is first compared against the repository's list of branches. If the text is a branch, then the user is prompted if they want to persist the branch for future commands, and only if the user responds affirmatively is the text stored in the `.cruft.json` file under the `branch` key.

During the execution of the `check` or `update` commands, if a `branch` key is present in the `.cruft.json` file, then the existence of the branch is checked against the repository's current branches. If the branch exists, the command continues as if the `checkout` option had been used with the branch as its value. If the branch is not in the repository's current branches, then the user is prompted to select from the available branches with the default noted.

**Comment by jch5f** [comment]
If folks think this would be a valuable enhancement, I am willing to take a crack at it.

**Comment by HappyEinara** [comment]
@samj1912 You're right that the issue you closed above is a dupe of this; apologies for overlooking this.

A little context to make the present issue clearer for future readers: persisting `checkout` in `.cruft.json` is now implemented by PR #80 (fixing related issue #92).

However the second part, for `check` and `update` to re-use the persisted `checkout` by default, is still open. This would be a breaking change, but it can still be overridden with an explicit `--checkout` and I'd be fairly surprised if anyone specifying an explicit template branch on `cruft create` would later want to check or update vs the template's default `main` branch. It would make template updates smoother in most cases IMO.

I note the labels Sambhav added so if I get time and it's not picked up I'll try to submit a PR at the weekend.

**Comment by adamcunnington-gdt** [comment]
I stumbled across the same thing.

It's silly that the checkout value persisted in cruft.json is not implicitly used. What is stopping this being fixed? It's quite an old issue now.

**Comment by clotildeguinard** [comment]
Hello! Before we can reuse the "checkout" entry from the cruft.json, there is another part that is missing, which is to also persist the "checkout" parameter when it is provided to the "link" command. PR #184 solves that.
